index.bs

```diff
@@ -5506,6 +5879,323 @@ partial interface MLGraphBuilder {
+        1. Make a request to the underlying platform to:
+            1. Let |opImpl| be an [=implementation-defined=] platform operator for the triangular operation, given |options|.
+            1. Store a reference of |opImpl| in |output|.{{MLOperand/[[operator]]}}.
+            1. Create an [=implementation-defined=] platform operand |outputImpl| to represent the output, given |output| and |opImpl|.
+            1. Store a reference to |outputImpl| in |output|.{{MLOperand/[[operand]]}}.
+        1. Connect |input|.{{MLOperand/[[operand]]}} as input to |opImpl|.
+        1. Connect |output|.{{MLOperand/[[operand]]}} as output to |opImpl|.
+    1. Return |output|.
+  </div>
+</details>
+
+<div class="example">
+<details open>
+  <summary>
+    Examples of how triangular works in different diagonal settings.
+  </summary>
+  <pre highlight="js">
+    // input:
+    //   [[7, 1, 2],
+    //    [9, 4, 8],
+    //    [2, 6, 3]]
+    const input = builder.constant(
+    { dimensions: [3,3] }, new Float32Array([7,1,2,9,4,8,2,6,3]));
+
+    // upper triangular matrix:
+    //   [[7, 1, 2], 
+    //    [0, 4, 8],
+    //    [0, 0, 3]]
+    const upper = builder.triangular(input);
+
+    // upper triangular matrix with one additional set of diagonals excluded:
+    //   [[0, 1, 2], 
+    //    [0, 0, 8],
+    //    [0, 0, 0]]
+    const upperPositive = builder.triangular(input, { diagonal: 1 });
+
+    // upper triangular matrix with one additional set of diagonals retained:
+    //   [[7, 1, 2], 
+    //    [9, 4, 8],
+    //    [0, 6, 3]]
+    const upperNegative = builder.triangular(input, { diagonal: -1 });
+
+    // lower triangular matrix:
+    //   [[7, 0, 0],
+    //    [9, 4, 0],
+    //    [2, 6, 3]]
+    const lower = builder.triangular(input, { upper: false });
+
+    // lower triangular matrix with one additional set of diagonals retained:
+    //   [[7, 1, 0],
+    //    [9, 4, 8],
+    //    [2, 6, 3]]
+    const lowerPositive = builder.triangular(input, { upper: false, diagonal: 1 });
+
+    // lower triangular matrix with one additional set of diagonals excluded:
+    //   [[0, 0, 0],
+    //    [9, 0, 0],
+    //    [2, 6, 0]]
+    const lowerNegative = builder.triangular(input, { upper: false, diagonal: -1 });
+  </pre>
+</details>
+</div>
+
+### where ### {#api-mlgraphbuilder-where}
+Select the values from the input or the other tensor depending on the corresponding {{boolean}} values of the condition tensor. The condition tensor is often the output of one of the element-wise logical operations.
+
+The input tensors will be broadcasted according to [[!numpy-broadcasting-rule]] to the final output shape. The [=rank=] of the output tensor is the maximum [=rank=] of the input tensors.
+For each dimension of the output tensor, its size is the maximum size along that dimension of the input tensors.
+
+<script type=idl>
+partial interface MLGraphBuilder {
+  MLOperand where(MLOperand condition, MLOperand input, MLOperand other);
```

Comment:
(feedback received in Chromium [CL-5039655](https://chromium-review.googlesource.com/c/chromium/src/+/5039655/comment/a746e5a8_8875fe4d/) review)

Recommend renaming arguments `input` to `trueValue` and `other` to `falseValue`. (however, folks agreed that naming is the hardest thing)

Reply:
Yeah, it's hard. I have no reason to update this PR with that proposed change b/c that same reason can be applied randomly elsewhere in any part of this .bs file.